onnx/defs/shape_inference.h

```diff
@@ -181,6 +181,19 @@ inline TensorShapeProto::Dimension operator*(TensorShapeProto::Dimension dim1, T
   return result;
 }
 
+template <typename Container>
+inline std::string stringify(const Container& elements) {
```

Comment:
Thanks for moving the implementation of getAttributeElementTypeAndLength into the .cc file. Minor nit: this template is part of the implementation and can also be inside the .cc file.